tests/test_type_alias_type.py

```diff
@@ -185,14 +185,15 @@ def test_type_alias_annotated_defs() -> None:
         'type': 'array',
         'minItems': 2,
         'prefixItems': [
-            {'$ref': '#/$defs/MyList_MaxLen_max_length_1_'},
-            {'$ref': '#/$defs/MyList_MaxLen_max_length_1_'},
+            {'$ref': '#/$defs/MyList_int__MaxLen_max_length_1_'},
+            {'$ref': '#/$defs/MyList_int__MaxLen_max_length_1_'},
         ],
         'maxItems': 2,
-        '$defs': {'MyList_MaxLen_max_length_1_': {'type': 'array', 'items': {'type': 'integer'}, 'maxItems': 1}},
+        '$defs': {'MyList_int__MaxLen_max_length_1_': {'type': 'array', 'items': {'type': 'integer'}, 'maxItems': 1}},
     }
 
 
+@pytest.mark.xfail(sys.version_info >= (3, 12), reason="TypeAliasType doesn't have __qualname__ yet")
```

Comment:
It fails for python _newer_ than (or equal to) 3.12? Saying "doesn't have qualname yet" makes me think the failure would happen for older-than-3.12, not 3.12 or newer.

Could you explain what's going wrong in a bit more detail?

Reply:
Nevermind I see this was just copying what existing tests were doing

Reply:
I see a way to resolve the need for these xfails, but I think that should be done in a separate PR, so ignore this comment.